src/report_modules/report_module_github_ci/src/report_format_github_ci.cpp

```diff
@@ -0,0 +1,230 @@
+            return -1;
+    }
+
+    std::string strFilepath = argv[1];
+    struct stat fileStatus;
+
+    XMLPlatformUtils::Initialize();
+
+    cParameterContainer inputParams;
+
+    // Default parameters
+    inputParams.SetParam("strInputFile", "Result.xqar");
+
+    if (StringEndsWith(ToLower(strFilepath), ".xqar"))
+    {
+        if (stat(strFilepath.c_str(), &fileStatus) == -1) // ==0 ok; ==-1 error
+        {
+            std::cerr << "Could not open file '" << strFilepath << "'!" << std::endl
+                      << "Abort generating report!" << std::endl;
+            return 1;
+        }
+
+        inputParams.SetParam("strInputFile", strFilepath);
+    }
+    else if (StringEndsWith(ToLower(strFilepath), ".xml"))
+    {
+        cConfiguration configuration;
+
+        std::cout << "Config: " << strFilepath << std::endl;
+        if (!cConfiguration::ParseFromXML(&configuration, strFilepath))
+        {
+            std::cerr << "Could not read configuration! Abort." << std::endl;
+            return -1;
+        }
+
+        inputParams.Overwrite(configuration.GetParams());
+
+        cConfigurationReportModule *reportModuleConfig = configuration.GetReportModuleByName(REPORT_MODULE_NAME);
+        if (nullptr != reportModuleConfig)
+            inputParams.Overwrite(reportModuleConfig->GetParams());
+        else
+            std::cerr << "No configuration for module '" << REPORT_MODULE_NAME << "' found. Start with default params."
+                      << std::endl;
+    }
+    else if (StringEndsWith(ToLower(strFilepath), "--defaultconfig"))
+    {
+        WriteDefaultConfig();
+        return 0;
+    }
+    else if (strcmp(strFilepath.c_str(), "-h") == 0 || strcmp(strFilepath.c_str(), "--help") == 0)
+    {
+        ShowHelp(strToolpath);
+        return 0;
+    }
+    else
+    {
+        ShowHelp(strToolpath);
+        XMLPlatformUtils::Terminate();
+        return -1;
+    }
+
+    // Exit with code 1 if an error was found to fail the GitHub pipeline
+    if(RunGithubCIReport(inputParams)) {
+        exit(1);
+    }
+}
+
+void ShowHelp(const std::string &toolPath)
+{
+    std::string applicationName = toolPath;
+    std::string applicationNameWithoutExt = toolPath;
+    GetFileName(&applicationName, false);
+    GetFileName(&applicationNameWithoutExt, true);
+
+    std::cout << "\n\nUsage of " << applicationNameWithoutExt << ":" << std::endl;
+    std::cout << "\nRun the application with xqar file: \n" << applicationName << " result.xqar" << std::endl;
+    std::cout << "\nRun the application with dbqa configuration: \n" << applicationName << " config.xml" << std::endl;
+    std::cout << "\nRun the application with and write default configuration: \n"
+              << applicationName << " --defaultconfig" << std::endl;
+    std::cout << "\n\n";
+}
+
+
+bool RunGithubCIReport(const cParameterContainer &inputParams)
+{
+    XMLPlatformUtils::Initialize();
+
+    auto *pResultContainer = new cResultContainer();
```

Comment:
Correct me if I'm wrong but I suspect this still creates a raw pointer, and now that delete has been removed we have also a memory leak.

I suggest to create a unique pointer here, since the iwnership is related to this function only, and this function transfer a temporary ownership to a sync PrintResult. We need the `<memory>` header.

```suggestion
    auto pResultContainer = std::maze_unique<cResultContainer>();
```

should do the job. Please notice that this signature changes: `PrintResults(std::unique_ptr<cResultContainer> & ptrResultContainer)`.

Reply:
I thought I fixed that. Probably an autorefactor oversight. I changed it now.